runelite-client/src/main/java/net/runelite/client/ui/overlay/components/TooltipComponent.java

```diff
@@ -51,6 +52,9 @@
 	@Setter
 	private Point position = new Point();
 
+	@Setter
+	private Client client;
```

Comment:
Passing here modIcons array instead of entire client would be better, as it is hard to tell why is entire client passed here just from looking at the component from outside, but with modIcons it would be obvious.